pkg/numaresourcesscheduler/objectstate/sched/sched_test.go

```diff
@@ -0,0 +1,98 @@
+package sched
+
+import (
+	nrsv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+)
+
+var dp = &appsv1.Deployment{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "test-deployent",
+		Namespace: "test-namespace",
+	},
+	Spec: appsv1.DeploymentSpec{
+		Template: corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Image: "quay.io/bar/image:v1",
+					},
+				},
+				Volumes: []corev1.Volume{
+					newSchedConfigVolume("foo", "bar"),
+				},
+			},
+		},
+	},
+}
+
+func TestUpdateDeploymentImageSettings(t *testing.T) {
+	type testCase struct {
+		imageSpec string
+	}
+
+	testCases := []testCase{
+		{
+			imageSpec: "quay.io/bar/image:v2",
+		},
+		{
+			imageSpec: "quay.io/bar/image:v3",
+		},
+		{
+			imageSpec: "quay.io/foo/image:v2",
+		},
+	}
+
+	podSpec := &dp.Spec.Template.Spec
+	for _, tc := range testCases {
+		UpdateDeploymentImageSettings(dp, tc.imageSpec)
+		if podSpec.Containers[0].Image != tc.imageSpec {
+			t.Errorf("failed to update deployemt image, expected: %q actual: %q", tc.imageSpec, podSpec.Containers[0].Image)
+		}
+	}
+}
+
+func TestUpdateDeploymentConfigMapSettings(t *testing.T) {
+	type testCase struct {
+		cmName string
+	}
+
+	testCases := []testCase{
+		{
+			cmName: "cm1",
+		},
+		{
+			cmName: "cm5",
+		},
+		{
+			cmName: "cm3",
+		},
+	}
+
+	podSpec := &dp.Spec.Template.Spec
+	for _, tc := range testCases {
+		UpdateDeploymentConfigMapSettings(dp, tc.cmName)
+		if podSpec.Volumes[0].Name != SchedulerConfigMapVolumeName {
+			t.Errorf("failed to update deployemt volume name, expected: %q actual: %q", SchedulerConfigMapVolumeName, podSpec.Volumes[0].Name)
+		}
+		if podSpec.Volumes[0].ConfigMap.LocalObjectReference.Name != tc.cmName {
+			t.Errorf("failed to update deployemt volume configmap name, expected: %q actual: %q", tc.cmName, podSpec.Volumes[0].ConfigMap.LocalObjectReference.Name)
+		}
+	}
+}
+
+func TestDeploymentNamespacedNameFromObject(t *testing.T) {
```

Comment:
non compulsory but perhaps the use of more meaningful variables will make the test easy to read (dp, nname)

Reply:
```dp``` is an abbreviate for deployment and ```nname``` for namespaced name. it's quite common actually and I saw that in used in multiple places